### PR TITLE
Allow hash_array to be findable in pandas 2.0; add workaround for test_resample for cudf.pandas

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
Fixes new failure in `test_hash_array`. Open to feedback on a better approach.

The main issue is that some public methods are defined under `__getattr__` with no `__dir__` to find them (which we rely on for module population)

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/pandas/_wrappers/pandas.py [hunk 1]
```diff
@@ -47,6 +48,22 @@
 cudf.set_option("mode.pandas_compatible", True)
 
 
+def _pandas_util_dir():
+    # In pandas 2.0, pandas.util contains public APIs under
+    # __getattr__ but no __dir__ to find them
+    # https://github.com/pandas-dev/pandas/blob/2.2.x/pandas/util/__init__.py
+    return list(importlib.import_module("pandas.util").__dict__.keys()) + [
```
Comment (vyasr) [inline python/cudf/cudf/pandas/_wrappers/pandas.py:55]: Can this just be
```suggestion
    return dir("pandas.util") + [
```
Then you could also drop importlib.

File: python/cudf/cudf_pandas_tests/test_cudf_pandas.py [hunk 2]
```diff
@@ -909,6 +909,9 @@ def test_resample():
     )
     expected = ser.resample("D").max()
     result = xser.resample("D").max()
+    # TODO: See if as_unit can be avoided
+    expected.index = expected.index.as_unit("s")
+    result.index = result.index.as_unit("s")
```
Comment (vyasr) [inline python/cudf/cudf_pandas_tests/test_cudf_pandas.py:914]: Are these fixes related to `hash_array`, are they other fixes needed for cudf.pandas, or are they completely unrelated? If the second, could you update the PR title/description accordingly? If the third, can we remove from this PR?

**Review by vyasr — APPROVED** [review]
I don't see any alternative to explicitly listing the attributes that are only visible via `pandas.util.__getattr__` calls. This is the only real solution I can think of.

File: python/cudf/cudf/pandas/_wrappers/pandas.py (hunk 1 above)
Comment (mroeschke) [reply python/cudf/cudf/pandas/_wrappers/pandas.py]: Ah thanks nice shortcut

**Comment by vyasr** [comment]
You're good to merge whenever from my end.

File: python/cudf/cudf/pandas/_wrappers/pandas.py (hunk 1 above)
Comment (mroeschke) [reply python/cudf/cudf/pandas/_wrappers/pandas.py]: Oh I remember why I didn't use `dir(pandas.util)`, the eventual assignment of this patch to `pd.util.__dir__` will cause a recursion error

**Review by galipremsagar — APPROVED** [review]

File: python/cudf/cudf/pandas/_wrappers/pandas.py (hunk 1 above)
Comment (vyasr) [reply python/cudf/cudf/pandas/_wrappers/pandas.py:55]: Ah good point, didn't think of that.

**Pull request merged**